File: tests/fixtures/edgecase_repo/lua/module5.lua
local mod4 = require("module4")

local Base5 = {}
Base5.__index = Base5

function Base5.new(name)
  local self = setmetatable({}, Base5)
  self.name = name
  return self
end

local Service5 = setmetatable({}, { __index = Base5 })

function Service5:handle(payload)
  self:validate(payload)
  return true
end

function Service5:validate(payload)
  return payload ~= nil
end

local function helper5(value)
  return value
end

return Service5

File: tests/fixtures/edgecase_repo/dart/module5.dart
import 'dart:async';

import 'module4.dart';

abstract class Handler5 {
  bool handle(String payload);
}

class Base5 {
  final String name;
  Base5(this.name);
}

class Service5 extends Base5 implements Handler5 {
  Service5(String name) : super(name);

  @override
  bool handle(String payload) {
    this.validate(payload);
    return true;
  }

  void validate(String payload) {}
}

String helper5(String value) => value;
